### DISCUSSION TITLE
Lottery: testing test_get_entrance_fee() fails

### DISCUSSION THREAD

**Discussion by pavelgor84** [smartcontractkit/full-blockchain-solidity-course-py#286] [Q&A]
Hi! I reached the first test for entrance fee function and have an error during testing this script:

**from brownie import Lottery, accounts, network, config
from web3 import Web3

def test_get_entrance_fee():
    account = accounts[0]
    lottery = Lottery.deploy(
        config["networks"][network.show_active()]["eth_usd_price_feed"],
        {"from": account},
    )

    assert lottery.getEntranceFee() > Web3.toWei(0.010, "ether")
    assert lottery.getEntranceFee() < Web3.toWei(0.013, "ether")**

ERROR:

>       assert lottery.getEntranceFee() > Web3.toWei(0.010, "ether")
E       AssertionError: assert 0 > 10000000000000000
E        +  where 0 = <ContractCall 'getEntranceFee()'>()
E        +    where <ContractCall 'getEntranceFee()'> = <Lottery Contract '0xE7eD6747FaC5360f88a2EFC03E00d25789F69291'>.getEntranceFee
E        +  and   10000000000000000 = <function to_wei at 0x7fd9e59173a0>(0.01, 'ether')
E        +    where <function to_wei at 0x7fd9e59173a0> = Web3.toWei

tests/test_lottery.py:12: AssertionError
============================================== short test summary info ==============================================
FAILED tests/test_lottery.py::test_get_entrance_fee - AssertionError: assert 0 > 10000000000000000
================================================= 1 failed in 9.38s 

It seems that test_get_entrance_fee() returns a number with floating point that raises the error " assert 0 > 10000000000000000".  But on the video this test is passing. 
My config:
etworks:
  mainnet-fork:
    eth_usd_price_feed: '0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419'

**Comment by cromewar** [comment]
Hello @pavelgor84 it seems your `getEntranceFee` function might be the error, if you check the return statement for the function is `uint256` it means it cannot have a floating point. As you can see it says 0 > 1000000000000000, this means your `getEntranceFee` is returning 0.. I have some recommendations for you:

1. Make sure your contract looks like this: Special attention to the commas on `(, int256 price, , , )`

```Solidity
import "@chainlink/contracts/src/v0.6/interfaces/AggregatorV3Interface.sol";
AggregatorV3Interface internal ethUsdPriceFeed;
  function getEntranceFee() public view returns (uint256) {
        (, int256 price, , , ) = ethUsdPriceFeed.latestRoundData();
        // set entrance fee to $50
        // Eth/Usd conversion has 8 decimals so in order to make it 18 (wei standard) it's necessary to multiply the price.
        uint256 adjustedPrice = uint256(price) * (10**10); //18 decimals
        uint256 costToEnter = (usdEntryFee * 10**18) / adjustedPrice;
        return costToEnter;
    }
```

2. Don't forget to setup the network properly with alchemy: [video timestamp 6:28:10](https://youtu.be/M576WGiDBdQ?t=23290) 
3. I'm going to suppose you just made a typo here:
![image](https://user-images.githubusercontent.com/20199180/139536919-e40336e2-c193-40a7-83d8-e79a9e9881a8.png)
nut make sure your `brownie-config.yaml` is correct.

I hope this info might help you.

**Reply by pavelgor84** [reply]
Thanks for answering, @cromewar !
Here my function :

import "@chainlink/contracts/src/v0.6/interfaces/AggregatorV3Interface.sol";
constructor(address _priceFeedAddress) public {
        uint256 usdEntryFee = 50 * (10**18);
        ethUsdPriceFeed = AggregatorV3Interface(_priceFeedAddress);
      }

    function getEntranceFee() public view returns (uint256) {
        (, int256 price, , , ) = ethUsdPriceFeed.latestRoundData();
        uint256 adjustedPrice = uint256(price) * 10**10; // 18 decimals (8 +10)
        // 50$, 2000 / ETH
        // 50 * 100000 / 2000
        uint256 costToEnter = (usdEntryFee * 10**18) / adjustedPrice;
        return costToEnter;
    }
And config file:
dependencies:
  - smartcontractkit/chainlink-brownie-contracts@1.1.1

compiler:
  solc:
    remappings:
      - '@chainlink=smartcontractkit/chainlink-brownie-contracts@1.1.1'
       
networks:
  mainnet-fork:
    eth_usd_price_feed: '0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419'

When I'm modifying this function to return just this int256 price, it shows me 436663268438. I made it via console. 
But when I change everything back, it shows me 0🤯

**Reply by cromewar** [reply]
I found the error and I have the explanation as well:  

Look this, you declare the variable `usdEntryFee`like this:

`uint256 usdEntryFee = 50 * (10**18);`

and then on your function you set `costToEnter` like this:

`uint256 costToEnter = (usdEntryFee * 10**18) / adjustedPrice;`

As you can see you are multiplying `usdEntryFee` twice which gives you a giant number I can't even describe.

So, uint256 has a maximum space capacity of 256 bytes, when you exceed that amount it would or restart or return 0, this is why you are getting the error.

Solution:

Change: `uint256 usdEntryFee = 50 * (10**18);`
to this: `uint256 public usdEntryFee;`

**Reply by pavelgor84** [reply]
I found an issue. "uint256 public usdEntryFeel;" has already been declared above the constructor and I added uint256 to this variable in the constructor. This caused an error!
Thanks @cromewar! You moved me on the right path!

**Reply by cromewar** [reply]
I'm glad it helped, remember don't multiply twice as well, I you have any other question feel free to ask.
